src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c

```diff
@@ -1001,6 +1046,23 @@ void xocl_subdev_destroy_by_level(xdev_handle_t xdev_hdl, int level)
 	xocl_unlock_xdev(xdev_hdl);
 }
 
+void xocl_subdev_destroy_by_level_slot(xdev_handle_t xdev_hdl, int level,
+		uint32_t slot_id)
```

Comment:
level is not needed here. The only ULP subdevs have valid slot_id.

Reply:
Actually, as slot id is unsigned int. Hence, the default value of all subdev is '0'. 
We may need this level also to protect this.

Reply:
I meant only ULP subdev has valid slot_id. So you may remove "int level" from the function argument list and hardcode ULP in the function if needed.